src/Coordination/KeeperStateMachine.cpp

```diff
@@ -105,6 +186,18 @@ void KeeperStateMachine::init()
         storage = std::make_unique<KeeperStorage>(coordination_settings->dead_session_check_period_ms.totalMilliseconds(), superdigest);
 }
 
+nuraft::ptr<nuraft::buffer> KeeperStateMachine::commit_ext(const nuraft::state_machine::ext_op_params & params)
+{
+    QueueParam param(params.log_idx, params.data);
+    while (!async_commit_queue.push(param))
```

Comment:
It break consensus guarantees after `commit` or `commit_ext` returned the request must be applied to the state machine.

Reply:
> It break consensus guarantees: after `commit` or `commit_ext` returned the request must be applied to the state machine otherwise nothing is guaranteed.

Notice that I only push the commit request to a queue which another thread will consume the queue and apply the log to state machine, then reply the result to client. From the perspective of client, it is no change, but the throughput of whole system is level up.

Reply:
> It break consensus guarantees: after `commit` or `commit_ext` returned the request must be applied to the state machine otherwise nothing is guaranteed.

Yes, you are right, this is not right for raft.